marginaleffects/plot.py

```diff
@@ -0,0 +1,193 @@
+            continue
+
+        if variable_type == 'numeric':
+            if key == first_key:
+                to_datagrid[key] = np.linspace(modeldata[key].min(), modeldata[key].max(), 100).tolist()
+            else:
+                to_datagrid[key] = np.percentile(modeldata[key], [0, 25, 50, 75, 100], method="midpoint").tolist()
+
+        elif variable_type == 'boolean' or variable_type == 'character':
+            to_datagrid[key] = modeldata[key].unique().to_list()
+            assert len(to_datagrid[key]) <= 10, f"Character type variables of more than 10 unique values are not supported. {ele} variable has {len(to_datagrid[ele])} unique values."
+
+
+
+    dt_code = "datagrid(newdata=modeldata"
+    for key, value in to_datagrid.items():
+        dt_code += ", " + key + "="
+        if isinstance(value, str):
+            dt_code += "'" + value + "'"
+        else:
+            dt_code += str(value)
+    dt_code += ")"
+
+    exec("global dt; dt = " + dt_code)
+
+    return dt
+
+
+def common_plot(dt, x_name, x_type, fig=None, axes_i=None, label=None, color=None):
+
+    x = dt.select(x_name).to_numpy().flatten()
+    y = dt.select("estimate").to_numpy().flatten()
+    y_std = dt.select("std_error").to_numpy().flatten()
+
+    if fig is not None:
+        plot_obj = fig.axes[axes_i] if axes_i is not None else plt
+    else:
+        fig = plt.figure()
+        plot_obj = plt
+    
+    if x_type == "numeric":
+        if color is None:
+            plot_obj.fill_between(x, y-y_std, y+y_std, alpha=0.2)
+            plot_obj.plot(x, y, label=label)
+        else:
+            plot_obj.fill_between(x, y-y_std, y+y_std, color=color, alpha=0.2)
+            plot_obj.plot(x, y, color=color, label=label)
+
+    elif x_type == "character" or x_type == "boolean":
+        if color is None:
+            plot_obj.errorbar(x, y, yerr=y_std, fmt='o', label=label)
+        else:
+            plot_obj.errorbar(x, y, yerr=y_std, fmt='o', color=color, label=label)
+
+    return fig
+
+
+def plot_predictions(
+    model,
+    condition=None,
+    by=False,
+    newdata=None,
+    vcov=True,
+    conf_level=0.95,
+    transform=None,
+    draw=True
+):
+
+    assert not (not by and newdata is not None), "The 'newdata' argument requires a 'by' argument"
+
+    assert (condition is None and by) or (condition is not None and not by), "One of the 'condition' and 'by' arguments must be supplied, but not both"
+
+    if by:
+        if isinstance(by, str):
+            by = [by]
+
+        if newdata is not None:
+            dt = predictions(model, newdata=newdata, conf_level=conf_level, vcov=vcov, transform=transform)
+        else:
+            dt = predictions(model, by=by, conf_level=conf_level, vcov=vcov, transform=transform)
+
+        dt = dt.sort(by[0])
+
+    if condition is not None:
+        built_dt = build_plot(model, condition)
```

Comment:
maybe rename `build_plot()` to something more explicit to indicate that this function is used only with the `condition` argument.